### PR TITLE
Fix TestServerAction for methods parameters

### PR DESCRIPTION
**Pull request by serweck**
This fix when you send a method like Guid.NewGuid() and execute this.

return await Fixture.Server .CreateHttpApiRequest<CommentsController>(r => r.GetComment(Guid.NewGuid(), id), new { version = 1 }) .WithIdentity(IdentitiesGenerator.Administrator, TestServerDefaults.AuthenticationScheme) .GetAsync();

### CODE REVIEW

**Comment by Sergio1192** [comment]
We need some tests to check for new features or bugs fixes :)

**Comment by serweck** [comment]
tests added :)

File: src/Acheve.TestHost/HttpResponseMessageExtensions.cs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using System.Text.Json;
 using System.Threading.Tasks;
```
Comment (evawythien) [inline src/Acheve.TestHost/HttpResponseMessageExtensions.cs:3]: test comment

File: src/Acheve.TestHost/HttpResponseMessageExtensions.cs [hunk 2]
```diff
@@ -20,5 +21,17 @@ public static async Task IsSuccessStatusCodeOrThrow(this HttpResponseMessage res
...
+        public static async Task<T> ReadContentAsAsync<T>(this HttpResponseMessage responseMessage)
+        {
+            var json = await responseMessage.Content.ReadAsStringAsync();
+
+            var options = new JsonSerializerOptions
+            {
+                PropertyNameCaseInsensitive = true
+            };
+
+            return JsonSerializer.Deserialize<T>(json, options);
```
Comment (Sergio1192) [inline src/Acheve.TestHost/HttpResponseMessageExtensions.cs:34]: We are using Newtonsoft.Json

Comment (Sergio1192) [inline src/Acheve.TestHost/HttpResponseMessageExtensions.cs:34]: We are using Newtonsoft.Json

File: src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs [hunk 3]
```diff
@@ -32,10 +32,31 @@ public void AddTokens<TController>(TestServerAction action, TestServerTokenColle
...
                     }
                 }
+                else if (parameters[i].ParameterType.IsArray
+                   && !IgnoreHeader(parameters[i]))
+                {
+                    dynamic tokenValue = action.ArgumentValues[i].Instance;
+
+                    if (tokenValue != null 
+                        && tokenValue.Length != 0 
+                        && (tokenValue[0].GetType().IsPrimitive
```
Comment (Sergio1192) [inline src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs:42]: Maybe you can use "parameters[i].ParameterType.GetElementType"

File: tests/UnitTests/Acheve.TestHost/Routing/Helpers/Person.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace UnitTests.Acheve.TestHost.Routing.Helpers;
+
+public class Person
```
Comment (Sergio1192) [inline tests/UnitTests/Acheve.TestHost/Routing/Helpers/Person.cs:3]: Better in folder "Models"

File: tests/UnitTests/Acheve.TestHost/Routing/Helpers/Person.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿namespace UnitTests.Acheve.TestHost.Routing.Helpers
+{
+    public class Person
```
Comment (Sergio1192) [inline tests/UnitTests/Acheve.TestHost/Routing/Helpers/Person.cs:3]: Better in folder "Models"

Comment (serweck) [reply tests/UnitTests/Acheve.TestHost/Routing/Helpers/Person.cs]: solved

File: src/Acheve.TestHost/HttpResponseMessageExtensions.cs (hunk 2 above)
Comment (serweck) [reply src/Acheve.TestHost/HttpResponseMessageExtensions.cs]: solved

File: src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs (hunk 3 above)
Comment (serweck) [reply src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs:42]: i can't get the type inside an array :(

imagen

File: tests/UnitTests/Acheve.TestHost/Routing/Models/Person.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿namespace UnitTests.Acheve.TestHost.Routing.Helpers
```
Comment (Sergio1192) [inline tests/UnitTests/Acheve.TestHost/Routing/Models/Person.cs:1]: Change the namespace

File: src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs (hunk 3 above)
Comment (Sergio1192) [reply src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs:42]: `		
int[] array = new int[] { 1, 2, 3};
		
Console.WriteLine(array.GetType()); // System.Int32[]
Console.WriteLine(array.GetType().GetElementType()); // System.Int32
Console.WriteLine(array.GetType().GetElementType() == typeof(int)); // True
`
This code works

Comment (Sergio1192) [reply src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs:42]: You can add the type check in a function to use it on line 18 as well

Comment (serweck) [reply src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs]: Done!
Thx for info and this page :)

File: tests/UnitTests/Acheve.TestHost/Routing/Models/Person.cs (hunk 6 above)
Comment (serweck) [reply tests/UnitTests/Acheve.TestHost/Routing/Models/Person.cs]: Done !

File: src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs [hunk 7]
```diff
@@ -15,27 +15,37 @@ public void AddTokens<TController>(TestServerAction action, TestServerTokenColle
...
+                        if (tokenValue != null)
+                        {
+                            tokens.AddToken(tokenName, tokenValue.ToString(), isConventional: false);
+                        }
+                    }
+                    else if (parameters[i].ParameterType.IsArray
+                       && IsPrimitiveType(parameters[i].ParameterType.GetElementType()))
                     {
-                        tokens.AddToken(tokenName, tokenValue.ToString(), isConventional: false);
+                        dynamic tokenValue = action.ArgumentValues[i].Instance;
```
Comment (Sergio1192) [inline src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs:33]: is "dynamic" mandatory? we try not to use it

Comment (serweck) [reply src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs]: it was not mandatory to use it, but I have given it a workarround

**Pull request merged**